src/Symfony/Component/Notifier/Bridge/GatewayApi/GatewayApiTransportFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\GatewayApi;
+
+use Symfony\Component\Notifier\Exception\IncompleteDsnException;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Piergiuseppe Longo <piergiuseppe.longo@gmail.com>
+ *
+ * @experimental in 5.3
+ */
+final class GatewayApiTransportFactory extends AbstractTransportFactory
+{
+    /**
+     * @return GatewayApiTransport
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+        $authToken = $dsn->getUser();
+        $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
+        $from = $dsn->getOption('from');
+        $port = $dsn->getPort();
+
+        if (!$from) {
+            throw new IncompleteDsnException('Missing from.', $dsn->getOriginalDsn());
+        }
+
+        if (!$authToken) {
+            throw new IncompleteDsnException('Missing auth token.', $dsn->getOriginalDsn());
+        }
+
+        if ('gatewayapi' === $scheme) {
+            return (new GatewayApiTransport($authToken, $from, $this->client, $this->dispatcher))->setHost($host)->setPort($port);
+        }
+
+        throw new UnsupportedSchemeException($dsn, 'gatewayapi', $this->getSupportedSchemes());
+    }
```

Comment:
Please pull up the scheme check like here, to first check if the scheme is supported and afterwards for missing options. Thanks

https://github.com/symfony/symfony/blob/7d846d3e82f4e3dabec7d4b5300f50a1dd9ed7d6/src/Symfony/Component/Notifier/Bridge/Sinch/SinchTransportFactory.php#L27-L47

Reply:
Done